ERCS/erc-7836.md

```diff
@@ -13,141 +13,221 @@ requires: 1193, 5792
+    prehash: boolean,
+    // Public key.
+    publicKey: `0x${string}`,
+    // Key type.
+    type: 'secp256k1' | 'p256' | 'webauthn-p256' | 'webcrypto-p256'
   },
-  chainId: '0x01',
-  signatureRequest: {
-    hash: '0x...'
-  }
-  capabilities: {
-    paymasterService: {
-      sponsor: 'My app'
-    }
-  }
+  // Digest of the call bundle to sign over.
+  digest: `0x${string}`
+  // JSON-RPC method version.
+  version: string
 }
-After calling `wallet_prepareCalls`, app developers are expected to sign the hash provided in the `signatureRequest` field and submit the prepared and signed calls back to the wallet (either via its [EIP-1193](./eip-1193.md) provider or to its server counterpart) using the second RPC method, `wallet_sendPreparedCalls`.
+```
+
+#### Example
+
+```typescript
+const response = await provider.request({
+  method: 'wallet_prepareCalls',
+  params: [{
+    calls: [{
+      to: '0xcafebabecafebabecafebabecafebabecafebabe',
+      data: '0xdeadbeef',
+    }],
+    capabilities: {
+      paymasterService: {
+        url: 'https://...',
+      },
+    },
+    chainId: '0x1',
+    key: {
+      prehash: false,
+      publicKey: '0xdeadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeef',
+      type: 'p256',
+    },
+    version: '1',
+  }]
+})
+/**
+ * {
+ *   capabilities: {
+ *     paymasterService: {
+ *       url: 'https://...'
+ *     }
+ *   },
+ *   chainId: '0x1',
+ *   context: { ... },
+ *   digest: '0x1234567890abcdef1234567890abcdef1234567890abcdef1234567890abcdef',
+ *   key: {
+ *     prehash: false,
+ *     publicKey: '0xdeadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeef',
+ *     type: 'p256',
+ *   },
+ *   version: '1',
+ * }
+ */
+```
 
 ### `wallet_sendPreparedCalls`
 
-Accepts prepared calls from the response to a `wallet_prepareCalls` request along with a signature, and returns an EIP-5792 call bundle ID.
+Instructs a Wallet to execute a prepared call bundle (response of `wallet_prepareCalls`) with an accompanying `signature`. 
 
 #### Parameters
 
-```typescript!
-type SendPreparedCallsParams = [{
-  version: string
-  type: string
-  data: any
-  chainId: `0x${string}`
-  signature: {
-    type: string
+> The request is identical to the response of `wallet_prepareCalls`, except that it includes a `signature`.
+
+```typescript
+type Request = {
+  method: 'wallet_sendPreparedCalls',
+  params: [{
+    // Capabilities.
+    capabilities: Record<string, any>
+    // Chain ID of the chain the calls are being submitted to.
+    chainId: `0x${string}`
+    // Data specific to the Wallet from the `wallet_prepareCalls` response.
+    // (e.g. ERC-4337 UserOperation or alternative).
+    context: unknown
+    // Key that was used to sign the call bundle.
+    key: {
+      // Whether the digest will be prehashed by the key (default behavior of WebCrypto APIs).
+      prehash?: boolean,
+      // Public key.
+      publicKey: `0x${string}`,
```

Comment:
What would be the value of publicKey for secp256k1? It's probably more ergonomic for both the app and the server wallet if they can receive an address. Would this get passed in this field or do we need another 'address' field for this case?

Reply:
Yep, I would assume an `address` type would be good. And aligns with the Sub Accounts ERC’s `key` field.